lib/backend.js

```diff
@@ -204,11 +204,17 @@ Backend.prototype.trigger = function(action, agent, request, callback) {
 // Submit an operation on the named collection/docname. op should contain a
 // {op:}, {create:} or {del:} field. It should probably contain a v: field (if
 // it doesn't, it defaults to the current version).
-Backend.prototype.submit = function(agent, index, id, op, options, callback) {
+Backend.prototype.submit = function(agent, index, id, op, options, originalCallback) {
+  var backend = this;
+  var request = new SubmitRequest(this, agent, index, id, op, options);
+
+  var callback = function(error, ops) {
+    backend.emit('after submit', error, request);
```

Comment:
Like we discussed, I think using an "-ing" suffix would help convey that this occurs while the submit is busy finishing up and not actually afterwards. So for example: 

```suggestion
    backend.emit('submit finalizing', error, request);
```

Reply:
I think I'm leaning towards `submit end` personally?

Reply:
TIL about "gerund"